Dockerfile

```
@@ -1,19 +1,51 @@
-#   Railway volume-backed single-tenant preview.
+#   Postgres-backed storage, hosted-single-tenant-volume for a volume-backed
+#   preview, or hosted-single-tenant-volume-sandboxed-railway for the explicit
+#   Railway Sandbox preview described in docs/reborn/railway-sandbox-operator.md.
 
 FROM node:22.23.1-bookworm-slim@sha256:813a7480f28fdadac1f7f5c824bcdad435b5bc1322a5968bbbdef8d058f9dff4 AS node_toolchain
 
+FROM debian:bookworm-slim@sha256:7b140f374b289a7c2befc338f42ebe6441b7ea838a042bbd5acbfca6ec875818 AS railway_cli
+
+ARG TARGETARCH
+ARG RAILWAY_CLI_VERSION=5.30.4
+RUN apt-get -o Acquire::Retries=3 update \
```

Hadolint (2.15.1)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)